### PR TITLE
Implement current state for chartconfig service

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#4040

### CODE REVIEW

File: pkg/v7/chartconfig/current.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+		return nil, nil
+
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, c.projectName),
+	}
```
Comment (rossf7) [inline pkg/v7/chartconfig/current.go]: This is the only change in behaviour from the current resource. The operator only considers chartconfig CRs managed by the operator. This was a fix @tuommaki made for the PM with the hundreds of configmaps causing memory issues.

In this case I think its also useful for the "feature branch" deploy of charts. Currently cluster-operator will delete charts manually created in the giantswarm namespace. cc @puja108 @indyfree

File: service/controller/kvm/v7/resource/chartconfig/current.go [hunk 2]
```diff
@@ -29,13 +30,23 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
...
+
+		return nil, nil
+	} else if guest.IsAPINotAvailable(err) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available")
 
 		// We can't continue without a successful K8s connection. Cluster
 		// may not be up yet. We will retry during the next execution.
-		reconciliationcanceledcontext.SetCanceled(ctx)
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
+		resourcecanceledcontext.SetCanceled(ctx)
```
Comment (xh3b4sd) [inline service/controller/kvm/v7/resource/chartconfig/current.go]: So why does the cancelation change now? I don't follow here.

Comment (rossf7) [reply service/controller/kvm/v7/resource/chartconfig/current.go]: My thinking was that the chartconfig resource is the final resource. So there is nothing after to stop. But I guess we could another resource later.

I reverted as its an unnecessary change. Also if we can't connect to the tenant there is nothing useful that can be done.

**Comment by rossf7** [comment]
@xh3b4sd PTAL. Should be good now.

File: pkg/v7/chartconfig/current.go (hunk 1 above)
Comment (indyfree) [reply pkg/v7/chartconfig/current.go]: Nice! Had the label fix tuomas made already in mind for my issue. But now already resolved itself. I should go on more holidays and hope my issue magically resolve itself ;) :+1:

**Pull request merged**